File: 1049-minimum-domino-rotations-for-equal-row/1049-minimum-domino-rotations-for-equal-row.py
class Solution:
    def minDominoRotations(self, tops: List[int], bottoms: List[int]) -> int:
        res = float('inf')
        for v in range(1, 7):
            topSwaps = bottomSwaps = 0
            valid = True

            for i in range(len(tops)):
                if tops[i] != v and bottoms[i] != v:
                    valid = False
                    break
                
                if tops[i] != v:
                    topSwaps += 1
                if bottoms[i] != v:
                    bottomSwaps += 1
            
            if valid:
                res = min(res, topSwaps, bottomSwaps)
        
        return -1 if res == float('inf') else res

File: 1049-minimum-domino-rotations-for-equal-row/1049-minimum-domino-rotations-for-equal-row.js
/**
 * @param {number[]} tops
 * @param {number[]} bottoms
 * @return {number}
 */
var minDominoRotations = function(tops, bottoms) {
    let res = Infinity;
    for (let v = 1; v <= 6; v++) {
        let topSwaps = 0, bottomSwaps = 0;
        let valid = true;

        for (let i = 0; i < tops.length; i++) {
            if (tops[i] !== v && bottoms[i] !== v) {
                valid = false;
                break
            }
            if (tops[i] !== v) topSwaps++;
            if (bottoms[i] !== v) bottomSwaps++;
        }
        if (valid) {
            res = Math.min(res, topSwaps, bottomSwaps);
        }
    }
    return res == Infinity ? -1 : res;
};